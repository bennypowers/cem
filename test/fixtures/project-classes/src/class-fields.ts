export class ClassFields {
  string: string;
  protected protected: string;
  private private: string;
  initialized = 'string';
  bool = false;
  num = 0;
  /**
   * The class is not a custom element, so even thought this field has a @property decorator,
   * this member entry mustn't have an attribute
   */
  @squishy
  @icky()
  @tickly({ giggly: true })
  @property({ attribute: 'multi-decorator', reflect: true }) decorated = 0;
  /** must be classified as a method */
  fieldMethod = () => {}
}
