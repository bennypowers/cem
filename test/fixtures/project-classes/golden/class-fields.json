{
  "schemaVersion": "1.0.0",
  "modules": [
    {
      "kind": "javascript-module",
      "path": "src/class-fields.js",
      "declarations": [
        {
          "name": "ClassFields",
          "members": [
            {
              "name": "string",
              "type": {
                "text": "string"
              },
              "kind": "field"
            },
            {
              "name": "protected",
              "type": {
                "text": "string"
              },
              "kind": "field",
              "privacy": "protected"
            },
            {
              "name": "private",
              "type": {
                "text": "string"
              },
              "kind": "field",
              "privacy": "private"
            },
            {
              "name": "initialized",
              "type": {
                "text": "string"
              },
              "default": "'string'",
              "kind": "field"
            },
            {
              "name": "bool",
              "type": {
                "text": "boolean"
              },
              "default": "false",
              "kind": "field"
            },
            {
              "name": "num",
              "type": {
                "text": "number"
              },
              "default": "0",
              "kind": "field"
            },
            {
              "name": "decorated",
              "type": {
                "text": "number"
              },
              "default": "0",
              "kind": "field"
            },
            {
              "name": "fieldMethod",
              "kind": "method"
            }
          ],
          "kind": "class"
        }
      ],
      "exports": [
        {
          "kind": "js",
          "name": "ClassFields",
          "declaration": {
            "name": "ClassFields",
            "module": "src/class-fields.js"
          }
        }
      ]
    }
  ]
}